lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -973,6 +973,21 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
     opts.ValVerMinor = (unsigned long)minor64;
   }
 
+  llvm::StringRef valSelectStr = Args.getLastArgValue(OPT_select_validator);
+  if (!valSelectStr.empty()) {
+    if (valSelectStr.equals_lower("auto")) {
+      opts.SelectValidator = ValidatorSelection::Auto;
+    } else if (valSelectStr.equals_lower("internal")) {
+      opts.SelectValidator = ValidatorSelection::Internal;
+    } else if (valSelectStr.equals_lower("external")) {
+      opts.SelectValidator = ValidatorSelection::External;
+    } else {
+      errors << "Unsupported value '" << valSelectStr
+             << "for -select-validator option.";
+      return 1;
```

Comment:
This can be written more cleanly as a string switch with a pattern something like:

This does require adding an "Invalid" case to `ValidatorSelection`.

```suggestion
   opts.SelectValidator = StringSwitch<ValidatorSelection>(valSelectStr)
     .Case("auto", ValidatorSelection::Auto)
     .Case("internal", ValidatorSelection::Internal)
     .Case("external", ValidatorSelection::External)
     .Default(ValidatorSelection::Invalid);
    if (opts.SelectValidator == ValidatorSelection::Invalid) {
      errors << "Unsupported value '" << valSelectStr
             << "for -select-validator option.";
      return 1;
    }
```

Reply:
Done.